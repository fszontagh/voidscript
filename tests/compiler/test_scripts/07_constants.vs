// Test constants and immutable values
const int $MAX_SIZE = 100;
const string $APP_NAME = "VoidScript Compiler Test";
const bool $DEBUG_MODE = true;
const float $PI = 3.14159;

printnl("Max size: ", $MAX_SIZE);
printnl("App name: ", $APP_NAME);
printnl("Debug mode: ", $DEBUG_MODE);
printnl("Pi value: ", $PI);

// Using constants in calculations
function calculateCircleArea(float $radius) float {
    return $PI * $radius * $radius;
}

float $area = calculateCircleArea(5.0);
printnl("Circle area (radius 5): ", $area);

// Constants in conditions
if ($DEBUG_MODE) {
    printnl("Debug information enabled");
} else {
    printnl("Debug information disabled");
}

// Using constants as array bounds
int[] $buffer = [];
for (int $i = 0; $i < 5; $i++) {
    $buffer[$i] = $i * 2;
}

printnl("Buffer contents:");
for (int $value : $buffer) {
    printnl("  ", $value);
}