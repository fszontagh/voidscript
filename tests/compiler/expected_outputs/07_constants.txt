Max size: 100
App name: VoidScript Compiler Test
Debug mode: true
Pi value: 3.14159
Circle area (radius 5): 78.5398
Debug information enabled
Buffer contents:
  0
  2
  4
  6
  8